pkg/apis/clusteroperator/validation/clusterversion.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/cluster-operator/pkg/apis/clusteroperator"
+)
+
+// ValidateClusterVersionName validates the name of a cluster version.
+var ValidateClusterVersionName = apivalidation.ValidateClusterName
```

Comment:
Is there any reason why the ClusterVersion name needs the extra restrictions impose on Cluster names?

Reply:
No probably not, just working off a bad example, I will remove this. Why do we do extra validation for clusters? I assume something dns related?

Reply:
That's a good question. The `ClusterName` field in `ObjectMeta` has that extra validation. I assumed that cluster-operator would reify a cluster with the same name as the name of the `Cluster` object, which would in turn populate the `ClusterName` field of objects in that cluster. However, it appears by the comments on `ClusterName`, that that field is not actually used.